packages/xwayland.rb

```diff
@@ -0,0 +1,174 @@
+require 'package'
+
+class Xwayland < Package
+  description 'X server configured to work with weston or sommelier'
+  homepage 'https://x.org'
+  version '1.20.9-1'
+  # Using xorg ToT due to large number of recent xwayland commits.
+  source_url 'https://github.com/freedesktop/xorg-xserver/archive/28ed4b95e90fb225fc32cfdce16df603ab600fcd.zip'
+  source_sha256 'e1311bdcaaf0fc61103804c5b4698679f4bac7a0dae97c2ac03dd32cb3cd0640'
+
+  
+  depends_on 'libepoxy'
+  depends_on 'xorg_proto'
+  depends_on 'libxtrans'
+  depends_on 'libxkbfile'
+  depends_on 'wayland'
+  depends_on 'eudev'
+  depends_on 'libxfont'
+  depends_on 'libbsd'
+  depends_on 'nettle'
+  depends_on 'libtirpc'
+  depends_on 'pixman'
+  depends_on 'xinit'
+  depends_on 'libxkbcommon'
+  depends_on 'libunwind'
+  depends_on 'font_util'
+  depends_on 'xorg_lib'
+  depends_on 'libtirpc'
+  depends_on 'font_util'
+  depends_on 'libbsd'
+  depends_on 'dbus'
+  depends_on 'lzma' => :build
+  depends_on 'xkbcomp'
+  depends_on 'glproto'
+  depends_on 'mesa'
+
+  # Patches are from Google xwayland overlay at https://source.chromium.org/chromiumos/chromiumos/codesearch/+/master:src/third_party/chromiumos-overlay/x11-base/xwayland/files/
+  def self.patch
+    url_patch1 = "https://chromium.googlesource.com/chromiumos/overlays/chromiumos-overlay/+/067ac4b5060c16e6687a97cbb4bcdbaf5a0b5639/x11-base/xwayland/files/0001-Eliminate-conflict-with-X11-Xlib.h-with-khronos-eglp.patch?format=TEXT"
+    uri_patch1 = URI.parse url_patch1
+    filename_patch1 = 'patch1_base64'
+    sha256sum_patch1 = '16f5d5d1986daf07b4e0f01eb0e68b53d8ef301d169f3cc9493158c8b4987f16'
+
+    if File.exist?(filename_patch1) && Digest::SHA256.hexdigest( File.read("./#{filename_patch1}") ) == sha256sum_patch1
+      puts "Unpacking patch1 source code".yellow
+    else
```

Comment:
Why is this `else` line here?  Not following.

Reply:
The rationale for this was that this might prevent redownload if doing a ```crew install -k``` followed by a ```crew install```, but I've taken out that logic, as it does make it cleaner.